lib/SIL/SILProfiler.cpp

```diff
@@ -465,21 +480,20 @@ struct PGOMapping : public ASTWalker {
       CounterMap[thenExpr] = NextCounter++;
       auto thenCount = loadExecutionCount(thenExpr);
       LoadedCounterMap[thenExpr] = thenCount;
-      if (auto elseExpr = IE->getElseExpr()) {
-        CounterMap[elseExpr] = parent;
-        auto count = loadExecutionCount(elseExpr);
-        if (!parent) {
-          auto thenVal = thenCount.getValue();
-          for (auto pCount = NextCounter - 1; pCount > 0; --pCount) {
-            auto cCount = LoadedCounts->Counts[pCount];
-            if (cCount > thenVal) {
-              count = cCount;
-              break;
-            }
+      auto elseExpr = IE->getElseExpr();
```

Comment:
I don't understand why this is being un-conditionalized; will there always be a non-null elseExpr now?

Reply:
Given a well-formed AST, the else subexpression of an IfExpr should always be non-null. The underlying bug here is that the coverage pass goes off and operates on incomplete ASTs.